docusaurus/docs/Android/03-ui/06-guides/08-adding-custom-attachments.mdx

```diff
@@ -0,0 +1,169 @@
+            .format(Date(it))
+        val attachment = Attachment(
+            type = "date",
+            extraData = mutableMapOf("payload" to date)
+        )
+        messageInputView.submitCustomAttachments(
+            attachments = listOf(attachment),
+            viewHolderFactory = DateAttachmentPreviewFactory()
+        )
+    }
+
+    dialogPickerDialog.show(requireActivity().supportFragmentManager, null)
+}
+```
+
+To keep it simple we are reusing the existing attachment button to show the date picker dialog.
+
+After the date has been selected, we create an instance of `Attachment` with the corresponding payload and submit it to the `MessageInputView`.
+
+Note that the `MessageInputView::submitCustomAttachments` method also requires a ViewHolder factory to provide support for displaying custom attachment previews in in the `MessageInputView`. Here's the code for this factory:
+
+```kotlin
+class DateAttachmentPreviewFactory : SelectedCustomAttachmentViewHolderFactory {
+    override fun createAttachmentViewHolder(
+        attachments: List<Attachment>,
+        parent: ViewGroup,
+    ): BaseSelectedCustomAttachmentViewHolder {
+        return ItemDateAttachmentPreviewBinding
+            .inflate(LayoutInflater.from(parent.context), parent, false)
+            .let(::DateAttachmentPreviewViewHolder)
+    }
+
+    class DateAttachmentPreviewViewHolder(
+        private val binding: ItemDateAttachmentPreviewBinding,
+    ) : BaseSelectedCustomAttachmentViewHolder(binding.root) {
+
+        override fun bind(attachment: Attachment, onAttachmentCancelled: (Attachment) -> Unit) {
+            binding.dateTextView.text = attachment.extraData["payload"].toString()
+            binding.deleteButton.setOnClickListener {
+                onAttachmentCancelled(attachment)
+            }
+        }
+    }
+}
+```
+
+Now you can send a custom attachment of type `date` to the chat. The resulting UI will look like this:
+
+![Sending Custom Attachment](../../assets/sending_custom_attachments.png)
+
+Next, you'll need to build a custom attachment factory to render the item in the message list.
+
+## Rendering Date Attachments
+
+To render a custom attachment in the [MessageListView](../04-message-components/02-message-list.mdx) you'll have to implement `AttachmentFactory` interface:
+
+```kotlin
+interface AttachmentFactory {
+    
+    fun canHandle(message: Message): Boolean
+    
+    fun createViewHolder(
+        message: Message,
+        listeners: MessageListListenerContainer?,
+        parent: ViewGroup,
+    ): InnerAttachmentViewHolder
+}
+```
+
+There are two methods that need to be implemented:
+
+* `canHandle`: Checks whether the factory can handle the given attachments.
+* ```
+* `createViewHolder`: Represents the attachment UI within the `MessageListView`.
+
+Let's see how to create an `AttachmentFactory` that is capable of handling date attachments:
+
+```kotlin
+class DateAttachmentFactory : AttachmentFactory {
```

Comment:
I think we are missing comments here.
It's easy for the reader to get paralyzed by a lot of code here, I think it would be more clear if we also provided short descriptions of what the various methods achieve.